File: typescript/libs/workspace/domain/system/src/lib/data/Extent.ts
import { ObjectType } from '@allors/workspace/meta/system';
import { Predicate } from './Predicate';
import { Sort } from './Sort';
import { IVisitable } from './visitor/IVisitable';

export interface Extent extends IVisitable {
  objectType: ObjectType;

  predicate?: Predicate;

  sorting?: Sort[];
}

File: dotnet/System/Workspace/Allors.Workspace.Domain/Data/Extent.cs
// <copyright file="IExtent.cs" company="Allors bv">
// Copyright (c) Allors bv. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Data
{
    using Meta;

    public interface Extent : IVisitable
    {
        IComposite ObjectType { get; }

        // TODO: move to Result
        Sort[] Sorting { get; set; }
    }
}
